power_grid_model_c/power_grid_model_cpp/include/power_grid_model_cpp/basics.hpp

```diff
@@ -0,0 +1,35 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#pragma once
+#ifndef POWER_GRID_MODEL_CPP_BASICS_HPP
+#define POWER_GRID_MODEL_CPP_BASICS_HPP
+
+#ifdef PGM_DLL_EXPORTS
+#error "You should not define PGM_DLL_EXPORTS when using the C++ headers"
```

Comment:
this sounds a little bit passive-agressive. maybe you can change it to

```suggestion
#error "Cannot export dynamic targets from the PGM C API wrapper. Please disable PGM_DLL_EXPORT."
```

Reply:
Done.